src/Bridge/LibraryBridgeHelper.cpp

```diff
@@ -61,26 +70,91 @@ void LibraryBridgeHelper::startBridge(std::unique_ptr<ShellCommand> cmd) const
 }
 
 
-bool LibraryBridgeHelper::initLibrary(const std::string & library_path, const std::string library_settings, const std::string attributes_names)
+bool LibraryBridgeHelper::checkBridgeIsRunning() const
 {
-    startBridgeSync();
-    auto uri = createRequestURI(LIB_NEW_METHOD);
+    String result;
+    try
+    {
+        ReadWriteBufferFromHTTP buf(createRequestURI(PING), Poco::Net::HTTPRequest::HTTP_GET, {}, ConnectionTimeouts::getHTTPTimeouts(getContext()));
+        readString(result, buf);
+    }
+    catch (...)
+    {
+        return false;
+    }
+
+    /*
+     * When pinging bridge we also pass current dicionary_id. The bridge will check if there is such
+     * dictionary. It is possible that such dictionary_id is not present only in two cases:
+     * 1. It is dictionary source creation and initialization of library handler on bridge side did not happen yet.
+     * 2. Bridge crashed or restarted for some reason while server did not.
+    **/
+    static constexpr auto dictionary_check = "dictionary=";
+    if (result.size() != (std::strlen(dictionary_check) + 1))
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Unexpected message from library bridge: {}. Check bridge and server have the same version.",
+                        result, std::strlen(dictionary_check));
+
+    UInt8 dictionary_id_exists;
+    auto parsed = tryParse<UInt8>(dictionary_id_exists, result.substr(std::strlen(dictionary_check)));
+    if (!parsed || (dictionary_id_exists != 0 && dictionary_id_exists != 1))
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Unexpected message from library bridge: {} ({}). Check bridge and server have the same version.",
+                        result, parsed ? toString(dictionary_id_exists) : "failed to parse");
+
+    LOG_TRACE(log, "dictionary_id: {}, dictionary_id_exists on bridge side: {}, library confirmed to be initialized on server side: {}",
+              toString(dictionary_id), toString(dictionary_id_exists), library_initialized);
+
+    if (dictionary_id_exists && !library_initialized)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Library was not initialized, but bridge responded to already have dictionary id: {}", dictionary_id);
+
+    if (!dictionary_id_exists && library_initialized)
+    {
+        LOG_WARNING(log, "Library bridge does not have library handler with dictionaty id: {}. It will be reinitialized.", dictionary_id);
```

Comment:
> It's unclear. What does it mean?

library_initialized - means server has received confirm that bridge initialized a library handler. But library bridge could crash, then need to reinitialize. In this 'if condition' i check for `library_initialized` to avoid initilizing as it has not even been initialized before.